main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -1295,13 +1295,26 @@ object Desugar {
 
   /**
     * Rewrites [[WeededAst.Expr.FCons]] (`x :: xs`) into a call to `List.Cons(x, xs)`.
+    * If there are over 20 literals we translate it to `Vector.toList(Vector#{1, 2, ...})`.
```

Comment:
I still think we should leave cons untouched - it's also redundant when we add the weeder recursion check

Reply:
Its a footgun for our users. What's the downside?